doc/essentials/EnergyPlusEssentials.tex

```diff
@@ -0,0 +1,2260 @@
+many measures are being considered, this duplicative editing can be
+inefficient and prone to errors. In addition, the measure cannot be
+easily applied to a different building energy model without again
+editing another set of files duplicating the original effort. Due
+to this, it is very common for some type of scripting to be used so
+that measures can be applied to the original building model and other
+models of other buildings reliably and quickly. Scripting in this
+approach means to apply some type of programming language to the task
+of modifying files.
+
+EnergyPlus includes, with the installation package, two methods of
+implementing measures EP-Macro and the ParametricPreprocessor. EP-Macro
+is similar to the C language pre-processor and allows for portions
+of the file to be included or excluded, portions of other files to
+be included when desired, and for specific entries that are normally
+fixed values to change programmatically. EP-Macro is documented in
+the AuxiliaryPrograms documentation. A file with lines starting ``\#\#''
+or with the ``imf'' file extension are likely to be a file using
+EP-Macro files. 
+
+The EnergyPlus ParametricPreprocessor uses special input objects in
+EnergyPlus to set values for any field in any other input object for
+a series of simple options. This is briefly described in the AuxiliaryPrograms
+documentation, and more details are present in the InputOutputReference
+under the section on ``Parametric Objects.'' The ParametricPreprocessor
+approach is very straight forward however it has limits in the flexibility
+that it provides. It is suitable for implementing measures related
+to internal loads, constructions, and simple efficiency changes but
+is probably not the appropriate tool for more complicated measures.
+
+Another approach to writing measures and using measures written by
+others is to use \href{https://www.openstudio.net/}{OpenStudio} and
+its \href{http://nrel.github.io/OpenStudio-user-documentation/}{documentation}.
+OpenStudio is closely aligned with EnergyPlus but has a different
+data model and file format. It is an application programming interface
+for EnergyPlus, and some graphical user interfaces are using it to
+work with EnergyPlus. It has two categories of measures: native OpenStudio
+measures and EnergyPlus measures. Both types of measures are usually
+written in the \href{https://www.ruby-lang.org/en/}{Ruby} programming
+language. Many measures are already available for OpenStudio and can
+be found in the \href{https://bcl.nrel.gov/}{Building Component Library.}It
+is an open source project. 
+
+Using the \href{https://www.python.org/}{Python} scripting language,
+\href{https://github.com/santoshphilip/eppy}{Eppy} allows access
+to any of the input objects in EnergyPlus as if they were native Python
+objects. It is very flexible and an especially good choice if you
+are already familiar with Python. It is also an open source project.
```

Comment:
Eppy isn't the only one -- https://github.com/Myoldmopar/py-idd-idf and probably [others](https://github.com/search?p=3&q=energyplus+python&type=Repositories)

Reply:
I added py-idd-idf and pyidf. I think those are the only other two that are meant to manipulate IDF.